src/pretix/control/forms/orders.py

```diff
@@ -118,3 +118,9 @@ class OrderContactForm(forms.ModelForm):
     class Meta:
         model = Order
         fields = ['email']
+
+
+class OrderLocaleForm(forms.ModelForm):
+    class Meta:
+        model = Order
+        fields = ['locale']
```

Comment:
Yes. The choices need to come from ``event.settings.locales``, which makes this a bit more complicated, you will probably need a custom ``__init__`` method on that form that sets the choices on the field (or creates the field dynamically).